src/typing_extensions.py

```diff
@@ -1199,12 +992,12 @@ def greet(name: str) -> None:
         At runtime this returns the first argument unchanged and otherwise
         does nothing.
         """
-        return __val
+        return val
 
 
-if hasattr(typing, "Required"):
+if hasattr(typing, "Required"):  # 3.11+
```

Comment:
These kinds of comments are very useful when doing this kind of PR, so I added them in various places that didn't have them